### DISCUSSION TITLE
property name for boolean values in ion-table #1003

### DISCUSSION THREAD

**Discussion by pedro-martins-brisa** [iurynogueira/ion#1010] [Ideas]
issue: #1003
pr: #1009

We are discussing the definition of a TypeScript interface, ColumnBooleanText, which represents text associated with boolean values. Code clarity and readability are crucial for effective development. Let's examine three naming options for the keys of this interface, and we would like to hear your opinions on which one you consider the best choice. 

![image](https://github.com/Brisanet/ion/assets/138057813/a1a9c5ad-f994-4269-95b8-2c529d486177)

## Option A:

```typescript
export interface ColumnBooleanText {
  on: string;
  off: string;
}
```

## Option B:

```typescript
export interface ColumnBooleanText {
  truthy: string;
  falsy: string;
}
```

## Option C:

```typescript
export interface ColumnBooleanText {
  positive: string;
  negative: string;
}
```

We'd like to know:

   - Which option do you believe is more descriptive and easy to understand?
   - Do you have any specific concerns about any of the options?
   - Is there any alternative naming you would suggest?

**Comment by iurynogueira** [comment]
To me, Option C is better, on/off (my suggestion) is very ugly hahaha

**Reply by danilo-moreira-brisa** [reply]
I prefer option C as well, I think is better to understand the purpose of these attributes.

**Reply by allan-chagas-brisa** [reply]
I'm also more inclined to C, like we discussed.

**Comment by vinicius-guedes-santos** [comment]
I think option "B" fits better.

I find it strange to use `positive` and `negative` in an interface called "ColumnBooleanText" kkkkkkkkk

It can imply that they are numeric values, and negative has ambiguities in the meaning:
![image](https://github.com/Brisanet/ion/assets/90062413/bb37dbbf-0e77-4df2-8bfd-0d9b8cc2f883)

`truthy` and `falsy` give me clarity that I am working with boolean values.

**Reply by pedro-martins-brisa** [reply]
Choosing option 'B' (truthy and falsy) makes sense, especially when we consider scenarios where the false value can be represented by different data types, such as undefined, null, 0, or ''(empty).

**Reply by allan-chagas-brisa** [reply]
But the property represents the text that will be displayed right? When you are naming a object key you have to consider its context, like this is the positive / negative column boolean text. We can't consider the key name as independent.

**Reply by vinicius-guedes-brisa** [reply]
when I said that `positive` and `negative` don't make sense with the interface name, I was analyzing the context, not the key independently

why would `positive` and `negative` be better than `truthy` and `falsy`?

**Reply by allan-chagas-brisa** [reply]
In my opinion truthy and falsy refers more to the boolean values than the text that will be displayed. But its fine, we can use them.
